src/main/java/com/coveros/selenified/element/check/Equals.java

```diff
@@ -102,17 +106,122 @@ int checkMatches(int expectedMatches, double waitFor, double timeTook) {
      * @return String: the actual tag name of the element. null will be returned if the element isn't present
      */
     String checkTagName(String expectedTagName, double waitFor, double timeTook) {
-        // get the actual class value
+        String check = this.element.prettyOutput() + " with tag name <b>" + expectedTagName + ENDB;
+        // get the actual tag name value
         String actualTagName = this.element.get().tagName();
         // record the result
-        if (expectedTagName == null ? actualTagName != null : !expectedTagName.equals(actualTagName)) {
-            this.reporter.fail(this.element.prettyOutput() + " with tag name <b>" + expectedTagName + ENDB, waitFor, this.element.prettyOutputStart() + " has a tag name of <b>" + actualTagName + ENDB, timeTook);
+        if (!this.element.is().present()) {
+            this.reporter.fail(check, waitFor, this.element.prettyOutputStart() + IS_NOT_PRESENT, timeTook);
+        } else if (!expectedTagName.equals(actualTagName)) {
+            this.reporter.fail(check, waitFor, this.element.prettyOutputStart() + " has a tag name of <b>" + actualTagName + ENDB, timeTook);
         } else {
-            this.reporter.pass(this.element.prettyOutput() + " with tag name <b>" + expectedTagName + ENDB, waitFor, this.element.prettyOutputStart() + " has a tag name of <b>" + expectedTagName + ENDB, timeTook);
+            this.reporter.pass(check, waitFor, this.element.prettyOutputStart() + " has a tag name of <b>" + expectedTagName + ENDB, timeTook);
         }
         return actualTagName;
     }
 
+    /**
+     * Checks that the element's location equals the provided expected location. If
+     * the element isn't present, this will constitute a failure, same as a
+     * mismatch. This information will be logged and recorded, with a screenshot
+     * for traceability and added debugging support.
+     */
+    abstract void location(Point expectedLocation);
+
+    /**
+     * Checks that the element's location equals the provided expected location. If
+     * the element isn't present, this will constitute a failure, same as a
+     * mismatch. This information will be logged and recorded, with a screenshot
+     * for traceability and added debugging support.
+     *
+     * @param expectedLocation - the full expected tag name
+     * @param waitFor          - if waiting, how long to wait for (set to 0 if no wait is desired)
+     * @param timeTook         - the amount of time it took for wait for something (assuming we had to wait)
+     * @return String: the actual tag name of the element. null will be returned if the element isn't present
+     */
+    Point checkLocation(Point expectedLocation, double waitFor, double timeTook) {
+        String check = this.element.prettyOutput() + " with location <b>" + expectedLocation.toString() + ENDB;
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal " with location <b>" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/sonar/coding_rules#rule_key=squid%3AS1192)